mod_app/views.py

```diff
@@ -15,10 +15,22 @@ class HomeView(TemplateView):
 class FilmListView(ListView):
     model = Film
     template_name = "film_list.html"
-    paginate_by = 20
+    paginate_by = 2
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        page = self.request.GET.get("list")
+
+        if page == "all":
+            context["object_list"] = context["film_list"] = self.model.objects.all()
+            context["is_paginated"] = False
+        elif page:
+            try:
+                self.paginate_by = int(page)
+            except ValueError:
```

Comment:
I don't this is doing what you're expecting it to do. self.paginate_by is the number of items to show per page, and it's already been used in super().get_context_data by the time you change it here.